src/EnergyPlus/SZVAVModel.cc

```diff
@@ -784,20 +788,32 @@ namespace SZVAVModel {
                     } else { // not enough capacity at this air flow rate. Unit does have enough capacity a full water/air, otherwise wouldn't be here
                         // this is different from the PTUnit and UnitarySys routines in this module
                         // find the water flow rate that meets the min load at region 1/2 bounday
-                        TempSolveRoot::SolveRoot(state, 0.001, MaxIter, SolFlag, lowWaterMdot, FanCoilUnits::CalcFanCoilWaterFlowResidual, 0.0, 1.0, Par);
-                        Par(9) = lowWaterMdot;
-                        if (SolFlag < 0) {
-                            MessagePrefix = "Step 2a: ";
+                        if (SZVAVModel.HCoilType_Num == HCoil_Water || !HeatingLoad) {
+                            TempSolveRoot::SolveRoot(state, 0.001, MaxIter, SolFlag, lowWaterMdot, FanCoilUnits::CalcFanCoilWaterFlowResidual, 0.0, 1.0, Par);
+                            Par(9) = lowWaterMdot;
+                            if (SolFlag < 0) {
+                                MessagePrefix = "Step 2a: ";
+                            } else {
+                                Par(9) = 0.0;
+                            }
+                            TempSolveRoot::SolveRoot(state, 0.001, MaxIter, SolFlag, PartLoadRatio, FanCoilUnits::CalcFanCoilAirAndWaterFlowResidual, 0.0, 1.0, Par);
+                            if (SolFlag < 0) {
+                                MessagePrefix = "Step 2b: ";
+                            }
                         } else {
-                            Par(9) = 0.0;
-                        }
-                        TempSolveRoot::SolveRoot(state, 0.001, MaxIter, SolFlag, PartLoadRatio, FanCoilUnits::CalcFanCoilAirAndWaterFlowResidual, 0.0, 1.0, Par);
-                        if (SolFlag < 0) {
-                            MessagePrefix = "Step 2b: ";
+                            TempSolveRoot::SolveRoot(state, 0.001, MaxIter, SolFlag, PartLoadRatio, FanCoilUnits::CalcFanCoilLoadResidual, 0.0, 1.0, Par);
+                            if (SolFlag < 0) {
+                                MessagePrefix = "Step 2: ";
+                            }
                         }
                     }
                 } else { // too much capacity when coil off, could lower air flow rate here to meet load if air flow is above minimum
-                    TempSolveRoot::SolveRoot(state, 0.001, MaxIter, SolFlag, PartLoadRatio, FanCoilUnits::CalcFanCoilAirAndWaterFlowResidual, 0.0, 1.0, Par);
+                    if (SZVAVModel.HCoilType_Num == HCoil_Water || !HeatingLoad) {
+                        TempSolveRoot::SolveRoot(
+                            state, 0.001, MaxIter, SolFlag, PartLoadRatio, FanCoilUnits::CalcFanCoilAirAndWaterFlowResidual, 0.0, 1.0, Par);
+                    } else {
+                        TempSolveRoot::SolveRoot(state, 0.001, MaxIter, SolFlag, PartLoadRatio, FanCoilUnits::CalcFanCoilLoadResidual, 0.0, 1.0, Par);
+                    }
```

Comment:
Added SolveRoot() function calling for electric heating coil part-load ratio calculation using a different residual function.